src/App.tsx

```diff
@@ -3,33 +3,50 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  movies: Movie[];
+  query: string;
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    movies: moviesFromServer,
+    query: '',
+  };
+
+  makeQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ query: event.target.value });
+  };
 
   render() {
+    const { query, movies } = this.state;
+
+    const visibleMovies = movies.filter(movie => (
```

Comment:
вынеси это в отдельный метод, не пиши много логики в render
```suggestion
    const visibleMovies = this.getVisibleMovies();
```

Reply:
make sense